Model/Service/Order.php

```diff
@@ -0,0 +1,176 @@
+ * If you are unable to obtain it through the world-wide-web, please send an email
+ * to support@buckaroo.nl so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade this module to newer
+ * versions in the future. If you wish to customize this module for your
+ * needs please contact support@buckaroo.nl for more information.
+ *
+ * @copyright Copyright (c) Buckaroo B.V.
+ * @license   https://tldrlegal.com/license/mit-license
+ */
+namespace Buckaroo\Magento2\Model\Service;
+
+use Buckaroo\Magento2\Model\ConfigProvider\Account;
+use Buckaroo\Magento2\Model\ConfigProvider\Factory;
+use Buckaroo\Magento2\Model\ConfigProvider\Method\Factory as MethodFactory;
+use Buckaroo\Magento2\Model\OrderStatusFactory;
+use Buckaroo\Magento2\Model\Method\Transfer;
+use Magento\Store\Api\StoreRepositoryInterface;
+use Buckaroo\Magento2\Logging\Log;
+use Magento\Sales\Model\ResourceModel\Order\CollectionFactory;
+use Buckaroo\Magento2\Helper\Data;
+
+class Order
+{
+    protected $accountConfig;
+    protected $configProviderMethodFactory;
+    protected $storeRepository;
+    protected $orderFactory;
+    protected $orderStatusFactory;
+    protected $helper;
+    protected $logging;
+
+    public function __construct(
+        Account $accountConfig,
+        MethodFactory $configProviderMethodFactory,
+        Factory $configProviderFactory,
+        StoreRepositoryInterface $storeRepository,
+        CollectionFactory $orderFactory,
+        OrderStatusFactory $orderStatusFactory,
+        Data $helper,
+        Log $logging
+    ) {
+        $this->accountConfig = $accountConfig;
+        $this->configProviderMethodFactory = $configProviderMethodFactory;
+        $this->configProviderFactory = $configProviderFactory;
+        $this->storeRepository = $storeRepository;
+        $this->orderFactory = $orderFactory;
+        $this->orderStatusFactory = $orderStatusFactory;
+        $this->helper = $helper;
+        $this->logging = $logging;
+    }
+
+    public function cancelExpiredTransferOrders()
+    {
+        if ($stores = $this->storeRepository->getList()) {
+            foreach ($stores as $store) {
+                $this->cancelExpiredTransferOrdersPerStore($store);
+            }
+        }
+        return $this;
+    }
+
+    protected function cancelExpiredTransferOrdersPerStore($store)
+    {
+        $this->logging->addDebug(__METHOD__ . '|1|' . var_export($store->getId(), true));
+        $statesConfig = $this->configProviderFactory->get('states');
+        $state = $statesConfig->getOrderStateNew($store);
+        if ($transferConfig = $this->configProviderMethodFactory->get('transfer')) {
+            if ($dueDays = abs($transferConfig->getDueDate())) {
+                $this->logging->addDebug(__METHOD__ . '|5|' . var_export($dueDays, true));
+                $orderCollection = $this->orderFactory->create()->addFieldToSelect(['*']);
+                $orderCollection
```

Comment:
manipulating data is part of the Model, which can support a Service class. Easier to read code logic and not get it bloated by composing the collection select and filters

Reply:
Magento doesn't limit this and such things can be in repositories, models, services
(example: Magento\Store\Model\Service\StoreConfigManager)
overall, 40 rows method doesn't look like bloated